src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -506,33 +501,35 @@ internal void ParseSolution()
             _defaultConfigurationName = null;
             _defaultPlatformName = null;
 
+            // Pool strings as we parse them, de-duplicating them in memory.
+            StringPool pool = new();
+
             // the raw list of project configurations in solution configurations, to be processed after it's fully read in.
-            Dictionary<string, string> rawProjectConfigurationsEntries = null;
+            Dictionary<ProjectConfigurationKey, string>? rawProjectConfigurationsEntries = null;
 
             ParseFileHeader();
 
-            string str;
-            while ((str = ReadLine()) != null)
+            while (TryReadLine(out ReadOnlySpan<char> line))
             {
-                if (str.StartsWith("Project(", StringComparison.Ordinal))
+                if (line.StartsWith("Project(".AsSpan(), StringComparison.Ordinal))
```

Comment:
The AsSpan() is a sign of a missing overload in .NET?

Reply:
I think the diff would be easier to read if you added an extension for this.

Reply:
An implicit conversion for string to `ReadOnlySpan<char>` was introduced in .NET Core 2.1.

https://learn.microsoft.com/en-us/dotnet/api/system.string.op_implicit?view=net-7.0

I pushed an update that adds extension methods for .NET Framework only to make the call sites cleaner. Thanks for the suggestion.